backend/src/rss.rs

```diff
@@ -0,0 +1,204 @@
+    creators: Vec<String>,
+    thumbnail_url: Option<String>, 
+    tracks: Vec<EventTrack>,
+}
+
+impl_from_db!(
+    Event,
+    select: {
+        events.{ id, title, description, creators, thumbnail, created, tracks },
+    },
+    |row| {
+        Self {
+            id: row.id(),
+            title: row.title(),
+            description: row.description(),
+            created: row.created(),
+            creators: row.creators(),
+            thumbnail_url: row.thumbnail(),
+            tracks: row.tracks(),
+        }
+    }
+);
+
+/// Generates the xml for an RSS feed of a series in Tobira.
+pub(crate) async fn generate_feed(context: &Arc<Context>, id: &str) -> Result<String, Response> {
+    let db_pool = &context.db_pool;
+    let tobira_url = context.config.general.tobira_url();
+    let series_link = format!("{tobira_url}/!s/{id}");
+    let rss_link = format!("{tobira_url}/~rss/series/{id}");
+
+    let Some(series_id) = Key::from_base64(id) else {
+        return Err(bad_request("invalid series ID"));
+    };
+
+    let db = db::get_conn_or_service_unavailable(db_pool).await?;
+
+    let query = "select opencast_id, title, description from series where id = $1";
+    let series_data = match db.query_one(query, &[&series_id]).await {
+        Ok(data) => data,
+        Err(_) => return Err(bad_request("DB error querying series data")),
+    };
+
+    let series_oc_id = series_data.get::<_, String>("opencast_id");
+    let series_title = series_data.get::<_, String>("title").clone();
+    let series_description = series_data
+        .get::<_, Option<String>>("description")
+        .unwrap_or_default();
+
+    let format = if cfg!(debug_assertions) {
+        ogrim::Format::Pretty { indentation: "  " }
+    } else {
+        ogrim::Format::Terse
+    };
+
+    let buf = xml!(
+        #[format = format]
+        <?xml version="1.0" encoding="UTF-8"?>
+        <rss
+            version="2.0"
+            xmlns:dc="http://purl.org/dc/elements/1.1/"
+            xmlns:content="http://purl.org/rss/1.0/modules/content/"
+            xmlns:atom="http://www.w3.org/2005/Atom"
+            xmlns:media="http://search.yahoo.com/mrss/"
+            xmlns:itunes="http://www.itunes.com/dtds/podcast-1.0.dtd"
+        >
+            <channel>
+                <title>{series_title}</title>
+                <link>{series_link}</link>
+                <description>{series_description}</description>
+                <language>"und"</language>
+                <itunes:category text="Education">
+                    <itunes:category text="Courses"/>
+                </itunes:category>
+                <itunes:explicit>"true"</itunes:explicit>
+                <itunes:image href={format!("{tobira_url}/~assets/logo-small.svg")} />
+                <atom:link href={rss_link} rel="self" type="application/rss+xml" />
+                {|buf| video_items(buf, &db, &series_oc_id, &series_title, &rss_link, tobira_url).await.unwrap()}
```

Comment:
`video_items` only returns an error if either there is any DB connection issue or our query is broken. I don't see how the latter should happen, so returning 500 is best in this case. And this is actually already done as panics are caught further up!

However, I would suggest replacing the `unwrap` anyway and making it a bit more explicit:

```suggestion
                {|buf| {
                    video_items(buf, &db, &series_oc_id, &series_title, &rss_link, tobira_url)
                        .await
                        .map_err(|e| {
                            error!("could not retrieve videos for RSS: {e}");
                            response::internal_server_error()
                        })?
                }}
```